### PR TITLE
Add context manager to id generator

### PR DESCRIPTION
**Pull request by cloutierMat**
## Motivation

Adding a context manager to allow for a streamlined usage of the id manager where a resource unsets after use.

### CODE REVIEW

**Review by pinzon — APPROVED** [review]
LGTM

File: localstack-core/localstack/utils/id_generator.py [hunk 1]
```diff
@@ -16,6 +17,12 @@ def set_custom_id_by_unique_identifier(self, unique_identifier: str, custom_id:
         with self._lock:
             self._custom_ids[unique_identifier] = custom_id
 
+    @contextmanager
+    def custom_id(self, resource_identifier: ResourceIdentifier, custom_id: str) -> None:
+        yield self.set_custom_id(resource_identifier, custom_id)
+
+        self.unset_custom_id(resource_identifier)
```
Comment (simonrw) [inline localstack-core/localstack/utils/id_generator.py:24]: We probably want to wrap this in a try/finally block to make sure that we reset the state if an error is thrown
Can we also cover this in the tests?

**Review by simonrw — APPROVED** [review]
Just a comment about catching any exceptions in the generator but otherwise LGTM!

File: localstack-core/localstack/utils/id_generator.py (hunk 1 above)
Comment (cloutierMat) [reply localstack-core/localstack/utils/id_generator.py:24]: Good thinking! I will add that

**Pull request merged**